### PR TITLE
Add draft label to collection name when needed on dashboard

### LINKED ISSUE #1049: When a collection is in draft state, include "Draft" after the collection title in the admin table on the dashboard

### PR DESCRIPTION
**Pull request by aaron-collier**
## Why was this change made?

Fixes #1049 

Screen Shot 2021-05-14 at 3 59 03 PM

Note: This is draft pending updated test(s).

### CODE REVIEW

File: spec/requests/dashboards_spec.rb [hunk 1]
```diff
@@ -174,28 +174,24 @@
...
-        <td><a href=\"#{collection_path(workful_collection)}\">MyString</a></td>
-        <td>5</td>
-        <td>1</td>
-        <td>1</td>
-        <td>1</td>
-        <td>1</td>
-        <td>1</td>
-        <td>Dec 02, 2020</td>
-      </tr>
+  <td><a href=\"#{collection_path(workful_collection)}\">MyString</a><strong><i></i></strong></td>
```
Comment (aaron-collier) [inline spec/requests/dashboards_spec.rb:177]: These two tests wouldn't pass without spacing these over after moving them into a component. Same below.

File: app/components/dashboard/all_collections_row_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<tr>
+  <td><%= link_to collection.head.name, collection %><strong><i><%= draft_label %></i></strong></td>
```
Comment (jcoyne) [inline app/components/dashboard/all_collections_row_component.html.erb:2]: Should there be a space before the draft_label?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: spec/requests/dashboards_spec.rb (hunk 1 above)
Comment (jcoyne) [reply spec/requests/dashboards_spec.rb:177]: I feel like checking for exact html in a request spec is a bit of an anti-pattern, but I don't think we need to fix it here as it was a pre-existing condition.

File: app/components/dashboard/all_collections_row_component.html.erb (hunk 2 above)
Comment (jcoyne) [reply app/components/dashboard/all_collections_row_component.html.erb:2]: Do we want to have `<strong><i></i></strong>` on the page even when there is no draft label?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/components/dashboard/all_collections_row_component.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<tr>
+  <td><%= link_to collection.head.name, collection %><% if draft? %><strong><i> <%= draft_label %></i></strong><% end %></td>
```
Comment (jcoyne) [inline app/components/dashboard/all_collections_row_component.html.erb:2]: How about putting all this HTML and the conditional into a method in the component?  so just `<= draft_label %>` here and then `draft_label` can `return unless draft?`.

File: app/components/dashboard/all_collections_row_component.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    attr_reader :collection, :counts
+
+    def draft?
+      return true if @collection.head&.state&.include? 'draft'
+
+      false
+    end
+
+    def draft_label
```
Comment (jcoyne) [inline app/components/dashboard/all_collections_row_component.rb:21]: why not return all the html from this method?

```ruby
  tag.strong do
   tag.i  ' - Draft'
  end
 ```

File: app/components/dashboard/all_collections_row_component.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      case @collection.head&.state
+      when 'first_draft'
+        return '- Draft'
```
Comment (jcoyne) [inline app/components/dashboard/all_collections_row_component.rb:24]: These returns aren't necessary if this is the last statement in the method. (e.g. get rid of the `nil`)

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/components/dashboard/all_collections_row_component.rb (hunk 4 above)
Comment (jcoyne) [reply app/components/dashboard/all_collections_row_component.rb:21]: better sill might be:
```ruby
tag.span ' - Draft', class: 'draft-tag'
```

and then add a `.draft-tag` css style.

Comment (aaron-collier) [reply app/components/dashboard/all_collections_row_component.rb:21]: This is what I was looking for! Thanks, I also wasn't happy with the HTML that way.

**Review by jcoyne — APPROVED** [review]

**Comment by jcoyne** [comment]
@aaron-collier looks like you need to clean up the typecheck (just set it to false).

**Pull request merged**